sql/item_subselect.cc

```diff
@@ -1248,8 +1266,17 @@ void Item_singlerow_subselect::reset()
 bool
 Item_singlerow_subselect::select_transformer(JOIN *join)
 {
+  /*
+    We cannot perform this transform for query preparation, for example
+    prepare s from 'select * from t6, v2 where ( select t6a*t6b=v2a ) != 0';
+    v2a is substituted during prepare, but being an outer reference, when
+    it is substituted again during the 1st execution, it's context is wrong.
+  */
+  if (!thd->is_first_query_execution())
+    return false;
+
   DBUG_ENTER("Item_singlerow_subselect::select_transformer");
-  if (changed)
+  if ((thd->stmt_arena->state == Query_arena::STMT_EXECUTED) && changed)
```

Comment:
we do not wish to run this transformer for the 2nd execution under any circumstances